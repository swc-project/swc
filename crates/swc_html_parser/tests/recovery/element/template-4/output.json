{
  "type": "Document",
  "span": {
    "start": 1,
    "end": 133
  },
  "mode": "no-quirks",
  "children": [
    {
      "type": "DocumentType",
      "span": {
        "start": 1,
        "end": 16
      },
      "name": "html",
      "publicId": null,
      "systemId": null,
      "raw": "<!doctype html>"
    },
    {
      "type": "Element",
      "span": {
        "start": 17,
        "end": 133
      },
      "tagName": "html",
      "namespace": "http://www.w3.org/1999/xhtml",
      "attributes": [
        {
          "type": "Attribute",
          "span": {
            "start": 23,
            "end": 32
          },
          "namespace": null,
          "prefix": null,
          "name": "lang",
          "rawName": "lang",
          "value": "en",
          "rawValue": "\"en\""
        }
      ],
      "children": [
        {
          "type": "Element",
          "span": {
            "start": 34,
            "end": 76
          },
          "tagName": "head",
          "namespace": "http://www.w3.org/1999/xhtml",
          "attributes": [],
          "children": [
            {
              "type": "Text",
              "span": {
                "start": 40,
                "end": 45
              },
              "data": "\n    ",
              "raw": "\n    "
            },
            {
              "type": "Element",
              "span": {
                "start": 45,
                "end": 68
              },
              "tagName": "title",
              "namespace": "http://www.w3.org/1999/xhtml",
              "attributes": [],
              "children": [
                {
                  "type": "Text",
                  "span": {
                    "start": 52,
                    "end": 60
                  },
                  "data": "Document",
                  "raw": "Document"
                }
              ],
              "content": null,
              "isSelfClosing": false
            },
            {
              "type": "Text",
              "span": {
                "start": 68,
                "end": 69
              },
              "data": "\n",
              "raw": "\n"
            }
          ],
          "content": null,
          "isSelfClosing": false
        },
        {
          "type": "Text",
          "span": {
            "start": 76,
            "end": 77
          },
          "data": "\n",
          "raw": "\n"
        },
        {
          "type": "Element",
          "span": {
            "start": 77,
            "end": 126
          },
          "tagName": "body",
          "namespace": "http://www.w3.org/1999/xhtml",
          "attributes": [],
          "children": [
            {
              "type": "Text",
              "span": {
                "start": 83,
                "end": 84
              },
              "data": "\n",
              "raw": "\n"
            },
            {
              "type": "Element",
              "span": {
                "start": 84,
                "end": 117
              },
              "tagName": "template",
              "namespace": "http://www.w3.org/1999/xhtml",
              "attributes": [],
              "children": [],
              "content": {
                "type": "DocumentFragment",
                "span": {
                  "start": 84,
                  "end": 117
                },
                "children": [
                  {
                    "type": "Text",
                    "span": {
                      "start": 94,
                      "end": 106
                    },
                    "data": "\n    \n",
                    "raw": "\n    \n"
                  }
                ]
              },
              "isSelfClosing": false
            },
            {
              "type": "Text",
              "span": {
                "start": 117,
                "end": 126
              },
              "data": "\n\n",
              "raw": "\n\n"
            }
          ],
          "content": null,
          "isSelfClosing": false
        }
      ],
      "content": null,
      "isSelfClosing": false
    }
  ]
}
