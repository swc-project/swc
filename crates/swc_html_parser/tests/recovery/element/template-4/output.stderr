  x Stray end tag "div"
   ,-[$DIR/tests/recovery/element/template-4/input.html:8:1]
 7 | <template>
 8 |     </div>
   :     ^^^^^^
 9 | </template>
   `----
